docs/docsite/rst/dev_guide/migrating_roles.rst

```diff
@@ -0,0 +1,345 @@
+|``ansible.module_utils`` in Python      + Use :abbr:`FQCN (Fully Qualified Collection Name)` for Python imports        |
++----------------------------------------+------------------------------------------------------------------------------+
+| Plugins in :file:`PLUGINTYPE_plugins/` + - Move to :file:`plugins/PLUGINTYPE/`  (such as :file:`plugins/filter/`)     |
+|                                        | - Use :abbr:`FQCN (Fully Qualified Collection Name)` external to collection  |
++----------------------------------------+------------------------------------------------------------------------------+
+| :file:`module_utils/`                  + - Move to :file:`plugins/module_utils/`                                      |
+|                                        | - Use :abbr:`FQCN (Fully Qualified Collection Name)` external to collection  |
++----------------------------------------+------------------------------------------------------------------------------+
+
+.. note::
+
+	 In standalone roles, some of the plugin directories referenced their plugin types in the plural sense; this is not the case in collections.
+
+Migrating a role to a collection
+================================
+
+To migrate from a standalone role to a collection role:
+
+
+1. Create a collection. You need a `Galaxy namespace <https://galaxy.ansible.com/docs/contributing/namespaces.html>`_ to import this collection to Galaxy.
+
+.. code-block:: bash
+
+  $ ansible-galaxy collection init mynamespace.mycollection
+
+This creates the collection directory structure.
+
+2. Copy the standalone role directory into the :file:`roles/` subdirectory of the collection.
+
+.. code-block:: bash
+
+  $ mkdir mynamespace/mycollection/roles/my_role/
+  $ cp -r /path/to/standalone/role/mynamespace/my_role/\* mynamespace/mycollection/roles/my_role/
+
+
+4. Move any modules to the :file:`plugins/modules/` directory.
+
+.. code-block:: bash
+
+  $ mv -r mynamespace/mycollection/roles/my_role/library/\* mynamespace/mycollection/plugins/modules/
+
+4. Move any other plugins to the appropriate :file:`plugins/PLUGINTYPE/` directory. The :file:`mynamespace/mycollection/plugins/README.md` file explains the types of plugins that the collection can contain within optionally created subdirectories.
+
+.. code-block:: bash
+
+  $ mv -r mynamespace/mycollection/roles/my_role/filter_plugins/\* mynamespace/mycollection/plugins/filter/
+
+5. Change any references to the role, modules, or plugins with the :abbr:`FQCN (Fully Qualified Collection Name)`.
+
+.. code-block:: yaml
+
+  ---
+  - name: example role by FQCN
+    hosts: some_host_pattern
+    tasks:
+      - name: import FQCN role from a collection
+        import_role:
+          name: mynamespace.mycollection.my_role
+
+Note in this example that the type of content from inside a collection is inferred contextually. See :ref:`collections_using_playbook` for more details.
```

Comment:
> inferred contextually

Not sure what is meant here? Does it just mean "when you call a role it will call a role, and when you call a module it will call a module?"

Not sure if this is the best way to say that, more like "use FQCN to call the roles / plugins / modules in your collection" or something like that. Or say nothing at all?

Also, for those (like me) who don't want to also have to rewrite all their playbooks after doing this (or just like using the shorthand instead of getting carpal tunnel from writing FQCNs on every task), what about showing the `collections:` keyword example as mentioned here: https://docs.ansible.com/ansible/latest/user_guide/collections_using.html#using-collections-in-a-playbook

```yaml
---
- name: example role by FQCN
  hosts: some_host_pattern
  collections:
    - mynamespace.mycollection
  tasks:
    - name: import role from a collection
      import_role:
        name: my_role
```